File: examples/bubbleSort.py
# バブルソート - 型ヒント付きバージョン
# Tsuchinoko変換用サンプル

def bubbleSort(lists: list[int]) -> tuple[list[int], int]:
    sorted_list: list[int] = list(lists)
    list_length: int = len(sorted_list)
    i: int = 0
    j: int = 0
    for i in range(list_length):
        for j in range(list_length - i - 1):
            if sorted_list[j] > sorted_list[j + 1]:
                temp: int = sorted_list[j]
                sorted_list[j] = sorted_list[j + 1]
                sorted_list[j + 1] = temp
    return sorted_list, list_length

def program_start() -> None:
    test_list: list[int] = [64, 34, 25, 12, 22, 11, 90]
    print("元のリスト:", test_list)
    sorted_lists, length = bubbleSort(test_list)
    print("ソート後のリスト:", sorted_lists)

if __name__ == "__main__":
    program_start()

File: examples/bubbleSort.rs
fn bubbleSort(lists: Vec<i64>) -> (Vec<i64>, i64) {
    let mut sorted_list: Vec<i64> = lists.clone();
    let list_length: i64 = (sorted_list.len() as i64);
    let i: i64 = 0;
    let j: i64 = 0;
    for i in 0..list_length {
        for j in 0..((list_length - i) - 1) {
            if (sorted_list[(j as usize)] > sorted_list[((j + 1) as usize)]) {
                let temp: i64 = sorted_list[(j as usize)];
                sorted_list[(j as usize)] = sorted_list[((j + 1) as usize)];
                sorted_list[((j + 1) as usize)] = temp;
            }
        }
    }
    return (sorted_list, list_length);
}
fn program_start() -> () {
    let mut test_list: Vec<i64> = vec![64, 34, 25, 12, 22, 11, 90];
    println!("{:?} {:?}", "元のリスト:", test_list);
    let (sorted_lists, length) : (_, _) = bubbleSort(test_list);
    println!("{:?} {:?}", "ソート後のリスト:", sorted_lists);
}
fn main() -> () {
    program_start();
}